### PR TITLE
build: compile using Angular 9

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -14,13 +14,6 @@
             "tsConfig": "tsconfig.lib.json"
           }
         },
-        "test": {
```
Comment (markwhitfeld) [inline angular.json:17]: Why did you remove the test configuration?

File: package.json [hunk 2]
```diff
@@ -77,8 +77,8 @@
     "build:integration": "ng run integration:build:production",
     "build:integration:ssr": "yarn build:integration && ng run integration:server:production",
     "// - TESTING": "Test all modules #requires yarn build:packages",
-    "test": "ng test",
```
Comment (markwhitfeld) [inline package.json:80]: Why are we moving away from `ng test`?

File: integration/main.server.ts [hunk 3]
```diff
@@ -1 +1,12 @@
...
+import { environment } from './environments/environment';
+
+if (environment.production) {
+  enableProdMode();
+}
+
 export { AppServerModule } from './app/app.server.module';
+export { ngExpressEngine } from '@nguniversal/express-engine';
+
+export { renderModule, renderModuleFactory } from '@angular/platform-server';
```
Comment (markwhitfeld) [inline integration/main.server.ts:12]: What is the reason for these exports and running the `enableProdMode();` in this file now?
Also, some of these exports are not used and can be removed.

File: integration/server.ts [hunk 4]
```diff
@@ -1,48 +1,79 @@
...
+// '__non_webpack_require__' is a proxy to Node 'require'
+// The below code is to ensure that the server is run only when not requiring the bundle.
+// eslint-disable-next-line
+declare const __non_webpack_require__: NodeRequire;
+// eslint-disable-next-line
+const mainModule = __non_webpack_require__.main;
+const moduleFilename = (mainModule && mainModule.filename) || '';
+if (moduleFilename === __filename || moduleFilename.includes('iisnode')) {
+  run();
+}
```
Comment (markwhitfeld) [inline integration/server.ts:77]: What is the need for the code above?
I don't see server.ts imported anywhere...
Or, is this from a generic template that you found (in that case you should leave it matching the generic template)

File: integration/server.ts [hunk 5]
```diff
@@ -1,48 +1,79 @@
...
 
-app.listen(PORT, () => {
-  console.log(`Express server is running and listening at http://localhost:${PORT}!`);
-});
+export * from './main.server';
```
Comment (markwhitfeld) [inline integration/server.ts:79]: Same question here, I don't see it imported anywhere...?

File: packages/hmr-plugin/src/tests/hmr-mock.ts [hunk 6]
```diff
@@ -82,17 +81,10 @@ export class AppMockModuleNoHmrLifeCycle {
...
 function createRootNode(selector = 'app-root'): void {
-  const document = TestBed.get(DOCUMENT);
-  const adapter: DomAdapter = new BrowserDomAdapter();
-
-  const root = adapter.firstChild(
-    adapter.content(adapter.createTemplate(`<${selector}></${selector}>`))
-  );
-
-  const oldRoots = adapter.querySelectorAll(document, selector);
-  oldRoots.forEach(oldRoot => adapter.remove(oldRoot));
```
Comment (markwhitfeld) [inline packages/hmr-plugin/src/tests/hmr-mock.ts:93]: So, we don't need to remove the old roots anymore?

File: packages/store/types/tsconfig.json [hunk 7]
```diff
@@ -19,7 +19,7 @@
     "typeRoots": ["node_modules/@types"],
     "baseUrl": ".",
     "paths": {
-      "@angular/*": ["../../../node_modules/@angular/*"],
+      "@angular/core": ["../../../node_modules/@angular/core/core.d.ts"],
```
Comment (markwhitfeld) [inline packages/store/types/tsconfig.json:22]: Interesting, what is the reason for this change?

File: tsconfig.build.json [hunk 8]
```diff
@@ -13,6 +13,7 @@
     }
   },
   "angularCompilerOptions": {
+    "enableIvy": false,
```
Comment (markwhitfeld) [inline tsconfig.build.json:16]: Is this needed so that our lib works for non-ivy too?

Comment (splincode) [reply tsconfig.build.json:16]: Yes, turn enableIvy only need after upgrade to Angular 10
 you will need to enable this and then, the library will be supported only by Angular 9 and Angular 10, and then the NGXS library will be supported only by Angular 9 and Angular 10

File: packages/store/types/tsconfig.json (hunk 7 above)
Comment (splincode) [reply packages/store/types/tsconfig.json:22]: It turns out that in reality the NGXS library uses only @angular/core in source code (use injectable, ngModule). Therefore, there is no need to specify any ng libraries. The most surprising surprise was that Angular now compiles its library files differently. They used to generate index.d.ts in the output. Therefore, everything worked for us. Now core.d.ts is generated (if it @angular/core), and for example testing.d.ts (if it @angular/core/testing).

File: packages/hmr-plugin/src/tests/hmr-mock.ts (hunk 6 above)
Comment (splincode) [reply packages/hmr-plugin/src/tests/hmr-mock.ts:93]: Yes, now Angular itself destroys the root.

File: integration/server.ts (hunk 5 above)
Comment (splincode) [reply integration/server.ts:79]: It needed for Angular Ivy and SSR + ViewEngine
Otherwise, it just doesn't start and doesn't work.

Official example:
https://github.com/angular/universal/blob/master/integration/hapi-engine-ve/server.ts#L69

Another examples:
https://github.com/Angular-RU/angular-universal-starter/blob/master/server.ts#L189
https://www.ganatan.com/tutorials/server-side-rendering-with-angular-universal

File: integration/server.ts (hunk 4 above)
Comment (splincode) [reply integration/server.ts:77]: See my comment below:
All new projects now look like this 

https://github.com/angular/universal/blob/master/integration/hapi-engine-ve/server.ts#L69
https://github.com/Angular-RU/angular-universal-starter/blob/master/server.ts#L189

File: integration/main.server.ts (hunk 3 above)
Comment (splincode) [reply integration/main.server.ts]: yes, you are right, enableProd not needed
but without the following imports, Angular SSR does not start

```ts
export { AppServerModule } from './app/app.server.module';

export { renderModule, renderModuleFactory } from '@angular/platform-server';
```

File: package.json (hunk 2 above)
Comment (splincode) [reply package.json:80]: Because `@angular-builders/jest` is not supported with Jest 26 and correctly Angular CLI 9, 
so you can just call the command:

```
$ jest --config jest.config.js 
```

File: angular.json (hunk 1 above)
Comment (splincode) [reply angular.json:17]: Because `@angular-builders/jest` is not supported with Jest 26 and correctly Angular CLI 9, 
so you can just call the command:

```
$ jest --config jest.config.js 
```

File: integrations/ivy/jest.config.js [hunk 9]
```diff
@@ -3,5 +3,10 @@ const path = require('path');
 module.exports = {
   displayName: 'ngxs-ivy',
   rootDir: path.resolve(),
-  testMatch: ['<rootDir>/src/**/**/*.spec.ts']
+  testMatch: ['<rootDir>/src/**/**/*.spec.ts'],
```
Comment (splincode) [inline integrations/ivy/jest.config.js:6]: It should be need for `angular-builders@9`

File: package.json (hunk 2 above)
Comment (splincode) [reply package.json]: good catch

File: angular.json (hunk 1 above)
Comment (splincode) [reply angular.json]: good catch
